src/app/interfaces/dynamic-form-schema.interface.ts

```diff
@@ -0,0 +1,35 @@
+import { FormArray } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { ChartSchemaNode } from 'app/interfaces/chart-release.interface';
+import { Option } from 'app/interfaces/option.interface';
+import { TreeNodeProvider } from 'app/modules/ix-forms/components/ix-explorer/ix-explorer.component';
+
+export interface DynamicFormSchema {
+  name: string;
+  description: string;
+  schema: DynamicFormSchemaNode[];
+}
+
+export interface DynamicFormSchemaNode {
+  variable: string;
+  type: string;
```

Comment:
Because different components support different options we'd need stronger typing than just string type. Discriminated union similar to `FieldConfig` should help. Currently `cidr`, `nodeProvider`, `options` and `private` are specific to certain node types only.